enterprise-suite/console_test/util/util.go

```diff
@@ -0,0 +1,189 @@
+package util
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const DefaultTimeout = 3 * time.Second
+
+type CmdBuilder struct {
+	name                 string
+	args                 []string
+	timeout              time.Duration
+	expectZeroExitStatus bool
+	panicOnError         bool
+	printStdout          bool
+	printStderr          bool
+	captureStdout        *strings.Builder
+	captureStderr        *strings.Builder
+}
+
+func Cmd(name string, args ...string) *CmdBuilder {
+	return &CmdBuilder{
+		name:                 name,
+		args:                 args,
+		timeout:              DefaultTimeout,
+		expectZeroExitStatus: true,
+		panicOnError:         false,
+	}
+}
+
+func (cb *CmdBuilder) Timeout(t time.Duration) *CmdBuilder {
+	cb.timeout = t
+	return cb
+}
+
+func (cb *CmdBuilder) NoTimeout() *CmdBuilder {
+	cb.timeout = 0
+	return cb
+}
+
+func (cb *CmdBuilder) AnyExitStatus() *CmdBuilder {
+	cb.expectZeroExitStatus = false
+	return cb
+}
+
+func (cb *CmdBuilder) PanicOnError() *CmdBuilder {
+	cb.panicOnError = true
+	return cb
+}
+
+func (cb *CmdBuilder) PrintStdout() *CmdBuilder {
+	cb.printStdout = true
+	return cb
+}
+
+func (cb *CmdBuilder) PrintStderr() *CmdBuilder {
+	cb.printStderr = true
+	return cb
+}
+
+func (cb *CmdBuilder) PrintOutput() *CmdBuilder {
+	cb.printStdout = true
+	cb.printStderr = true
+	return cb
+}
+
+func (cb *CmdBuilder) CaptureStdout(out *strings.Builder) *CmdBuilder {
+	cb.captureStdout = out
+	return cb
+}
+
+func (cb *CmdBuilder) CaptureStderr(out *strings.Builder) *CmdBuilder {
+	cb.captureStderr = out
+	return cb
+}
+
+func (cb *CmdBuilder) Run() (int, error) {
+	var cmd *exec.Cmd
+
+	// Set up timeout context if needed
+	if cb.timeout == 0 {
+		cmd = exec.Command(cb.name, cb.args...)
+	} else {
+		ctx, cancel := context.WithTimeout(context.Background(), cb.timeout)
+		cmd = exec.CommandContext(ctx, cb.name, cb.args...)
+		defer cancel()
+	}
+
+	var exitcode int
+	var cmderr error
+
+	// Setup stdout/sterr pipes
+	cmdstderr, err := cmd.StderrPipe()
+	if err != nil {
+		exitcode, cmderr = -100, err
+	}
+	cmdstdout, err := cmd.StdoutPipe()
```

Comment:
`StdoutPipe`/`StderrPipe` docs say it's an error to call `Wait` before consuming the pipe. But it looks like we only consume these when we have nonempty `stdoutWriters`/`stderrWriters`. So we shouldn't create these unless we are going to consume them (or always consume below - I think it will be a noop if you do `io.MultiWriter(nil)`).

Reply:
Right, restructured this part.